ui/brave_custom_notification/message_view.h

```diff
@@ -0,0 +1,178 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_UI_BRAVE_CUSTOM_NOTIFICATION_VIEWS_MESSAGE_VIEW_H_
+#define BRAVE_UI_BRAVE_CUSTOM_NOTIFICATION_VIEWS_MESSAGE_VIEW_H_
+
+#include <memory>
+
+#include "base/macros.h"
+#include "base/observer_list.h"
+#include "base/strings/string16.h"
+#include "third_party/skia/include/core/SkBitmap.h"
+#include "third_party/skia/include/core/SkColor.h"
+#include "ui/gfx/geometry/insets.h"
+#include "ui/gfx/image/image.h"
+#include "ui/gfx/image/image_skia.h"
+#include "brave/ui/brave_custom_notification/public/cpp/notification.h"
+#include "brave/ui/brave_custom_notification/public/cpp/notification_delegate.h"
+#include "ui/views/animation/ink_drop_host_view.h"
+#include "ui/views/animation/slide_out_controller.h"
+#include "ui/views/animation/slide_out_controller_delegate.h"
+#include "ui/views/controls/focus_ring.h"
+#include "ui/views/focus/focus_manager.h"
+#include "ui/views/view.h"
+
+namespace views {
+class ScrollView;
+}  // namespace views
+
+namespace brave_custom_notification {
+
+class Notification;
+class NotificationControlButtonsView;
+
+class MessageView 
+    : public views::InkDropHostView,
+      public views::SlideOutControllerDelegate,
+      public views::FocusChangeListener {
+ public:
+  explicit MessageView(const Notification& notification);
+  class Observer {
+   public:
+    virtual ~Observer() = default;
+
+    virtual void OnSlideStarted(const std::string& notification_id) {}
+    virtual void OnSlideChanged(const std::string& notification_id) {}
+    virtual void OnPreSlideOut(const std::string& notification_id) {}
+    virtual void OnSlideOut(const std::string& notification_id) {}
+    virtual void OnCloseButtonPressed(const std::string& notification_id) {}
+  };
+
+  enum class Mode {
+    // Normal mode.
+    NORMAL = 0,
+    // "Pinned" mode flag. This mode is for pinned notification.
+    // When this mode is enabled:
+    //  - Swipe: partially possible, but limited to the half position
+    //  - Close button: hidden
+    //  - Settings and snooze button: visible
+    PINNED = 1,
+    // "Setting" mode flag. This mode is for showing inline setting panel in
+    // the notification view.
+    // When this mode is enabled:
+    //  - Swipe: prohibited
+    //  - Close button: hidden
+    //  - Settings and snooze button: hidden
+    SETTING = 2,
```

Comment:
Are these modes required for ad notifications

Reply:
Removed